### PR TITLE
Update compiling_workflows.rst

### PR DESCRIPTION
**Pull request by josh146**
Minor type correction in the compilation quickstart

### CODE REVIEW

**Review by albi3ro — APPROVED** [review]

File: doc/introduction/compiling_workflows.rst [hunk 1]
```diff
@@ -61,7 +61,7 @@ models:
 
     import jaxopt
 
-    @jax.jit
+    @qml.qjit
```
Comment (dime10) [inline doc/introduction/compiling_workflows.rst:64]: Just confirming if the change was tested? :)

**Review by dime10 — APPROVED** [review]

File: doc/introduction/compiling_workflows.rst (hunk 1 above)
Comment (maliasadi) [inline doc/introduction/compiling_workflows.rst:64]: I don't think this is correct!

**Review by maliasadi — CHANGES_REQUESTED** [review]

File: doc/introduction/compiling_workflows.rst (hunk 1 above)
Comment (dime10) [reply doc/introduction/compiling_workflows.rst]: Why is that the case?

File: doc/introduction/compiling_workflows.rst [hunk 2]
```diff
@@ -76,6 +85,10 @@ models:
 
         return params
 
+Compiling the entire hybrid workflow using ``@qjit`` however will lead to better
```
Comment (maliasadi) [inline doc/introduction/compiling_workflows.rst:88]:
```suggestion
Compiling the entire hybrid workflow using ``@qml.qjit`` however will lead to better
```

File: doc/introduction/compiling_workflows.rst [hunk 3]
```diff
@@ -45,17 +45,26 @@ compile your hybrid workflows:
...
+.. code-block:: python
+
+    @qml.qjit
+    def hybrid_function(params, x):
+        grad = qml.grad(circuit)(params)[0]
+        return jnp.abs(grad - x) ** 2
+
+In addition, functions that are compiled with ``@jax.jit`` can contain calls
+to qjit-compiled functions. For example, below we compile the full optimization loop,
+using ``@jax.jit``, for example when training models:
```
Comment (maliasadi) [inline doc/introduction/compiling_workflows.rst:67]: @josh146 line 79, please change `cost` to `circuit`?

Comment (maliasadi) [reply doc/introduction/compiling_workflows.rst:67]: It's really bothering that you can add any comments to that line in GH

Comment (josh146) [reply doc/introduction/compiling_workflows.rst:67]: done!

**Review by maliasadi — APPROVED** [review]
:tada:

File: doc/introduction/compiling_workflows.rst (hunk 3 above)
Comment (josh146) [inline doc/introduction/compiling_workflows.rst:67]:
```suggestion
In addition, functions that are compiled with ``@jax.jit`` can contain calls
to qjit-compiled functions. For example, below we compile a full optimization loop,
using ``@jax.jit``:
```

**Pull request merged**